pkg/userop/testing/local_blockchain/node.go

```diff
@@ -0,0 +1,303 @@
+				"--noEthCallOverrideSupport", "true",
+				"--useUserOperationGasLimitsForSubmission", "true",
+			},
+			ImagePlatform: "linux/amd64",
+			ExposedPorts:  []string{fmt.Sprintf("%s:%s/tcp", port, port)},
+			WaitingFor:    wait.ForLog("Server listening at"),
+		},
+		Started: true,
+	})
+	require.NoError(t, err, "failed to start Alto bundler container")
+
+	containerPort, err := altoContainer.MappedPort(ctx, port)
+	require.NoError(t, err, "failed to get Alto bundler container port")
+	bundlerURL, err := url.Parse(fmt.Sprintf("http://0.0.0.0:%s", containerPort.Port()))
+
+	bundlerActiveNode = &BundlerNode{
+		Container:    altoContainer,
+		ContainerURL: bundlerURL,
+	}
+
+	t.Cleanup(bundlerNodeClenaupFactory(ctx, t))
+	bundlerActiveUsers++
+
+	require.NoError(t, err, "failed to parse local bundler URL")
+
+	return bundlerActiveNode.ContainerURL
+}
+
+type Contracts struct {
+	EntryPoint          common.Address
+	Validator           common.Address
+	Factory             common.Address
+	Logic               common.Address
+	Paymaster           common.Address
+	SessionKeyValidator common.Address
+}
+
+var (
+	cachedContracts *Contracts
+	contractsMutex  sync.Mutex
+)
+
+func SetupContracts(ctx context.Context, t *testing.T, node *EthNode) Contracts {
+	contractsMutex.Lock()
+	defer contractsMutex.Unlock()
+
+	if cachedContracts != nil {
+		return *cachedContracts
+	}
+	chainID, err := node.Client.ChainID(ctx)
+	require.NoError(t, err)
+	slog.Info("chainID", "chainID", chainID)
+
+	balance := decimal.NewFromFloat(50e18 /* 50 ETH */).BigInt()
+	owner, err := NewAccountWithBalance(ctx, balance, node)
+	require.NoError(t, err, "failed to create owner account")
+
+	entryPoint, _, _, err := entry_point_v0_6_0.DeployEntryPoint(owner.TransactOpts, node.Client)
+	require.NoError(t, err)
+	slog.Info("deployed EntryPoint contract", "address", entryPoint)
+
+	validator, _, _, err := kernel_ecdsa_validator_v2_2.DeployKernelECDSAValidator(owner.TransactOpts, node.Client)
+	require.NoError(t, err)
+	slog.Info("deployed KernelECDSAValidator contract", "address", validator)
+
+	logic, _, _, err := kernel_v2_2.DeployKernel(owner.TransactOpts, node.Client, entryPoint)
+	require.NoError(t, err)
+	slog.Info("deployed Kernel contract", "address", logic)
+
+	factory, _, FactoryContract, err := kernel_factory_v2_2.DeployKernelFactory(owner.TransactOpts, node.Client, owner.Address, entryPoint)
+	require.NoError(t, err)
+	slog.Info("deployed KernelFactory contract", "address", factory)
+
+	tx, err := FactoryContract.SetImplementation(owner.TransactOpts, logic, true)
+	require.NoError(t, err)
+	slog.Info("set Kernel implementation", "tx", tx.Hash().Hex())
+
+	var paymaster common.Address // TODO: deploy Paymaster contract
+
+	sessionKeyValidator, _, _, err := session_key_validator.DeploySessionKeyValidator(owner.TransactOpts, node.Client)
+	require.NoError(t, err)
+	slog.Info("deployed SessionKeyValidator contract", "address", sessionKeyValidator)
```

Comment:
Rename `session_key_validator` to `session_key_validator_v2_4` and put session key validator deployment right after ecdsa validator deployment, so the workflow reflects the order in `Contracts` struct.